### PR TITLE
done

### PR DESCRIPTION
**Pull request by khorkovenko**

### CODE REVIEW

File: store/pom.xml [hunk 1]
```diff
@@ -67,6 +67,11 @@
 			<artifactId>spring-boot-starter-web</artifactId>
 			<version>3.1.5</version>
 		</dependency>
+		<dependency>
+			<groupId>org.springframework.data</groupId>
+			<artifactId>spring-data-jpa</artifactId>
+			<version>3.1.5</version>
+		</dependency>
```
Comment (sokimaaa) [inline store/pom.xml]: you should use starters when work with spring boot

File: store/pom.xml [hunk 2]
```diff
@@ -67,6 +67,11 @@
 			<artifactId>spring-boot-starter-web</artifactId>
 			<version>3.1.5</version>
```
Comment (sokimaaa) [inline store/pom.xml]: No need to specify versions of starters. Spring boot manages this.
```suggestion
```

File: store/src/main/java/com/book/store/repo/impl/BookRepositoryImpl.java [hunk 3]
```diff
@@ -2,51 +2,12 @@
...
-    public Optional<Book> findById(Long id) {
-        try (Session session = sessionFactory.getCurrentSession()) {
-            Book book = session.get(Book.class, id);
-            return Optional.ofNullable(book);
-        } catch (PersistenceException e) {
-            throw new RuntimeException("Error finding book by id: " + id, e);
-        }
-    }
+@Transactional
+public interface BookRepositoryImpl extends JpaRepository<Book, Long>, BookRepository {
```
Comment (sokimaaa) [inline store/src/main/java/com/book/store/repo/impl/BookRepositoryImpl.java]: you dont need this anymore
```suggestion
```

File: store/src/main/java/com/book/store/service/impl/BookServiceImpl.java [hunk 4]
```diff
@@ -41,10 +40,37 @@ public BookDto getBookById(Long id) {
...
+    public BookDto updateBook(Long id, BookRequestDto updateBookDto) {
+        Book existingBook = bookRepository.findById(id)
+                .orElseThrow(()
+                        -> new EntityNotFoundException("Book with id " + id + " not found."));
+
+        existingBook.setTitle(updateBookDto.getTitle());
+        existingBook.setAuthor(updateBookDto.getAuthor());
+        existingBook.setIsbn(updateBookDto.getIsbn());
+        existingBook.setPrice(updateBookDto.getPrice());
+        existingBook.setDescription(updateBookDto.getDescription());
```
Comment (Rommelua) [inline store/src/main/java/com/book/store/service/impl/BookServiceImpl.java]: you can use mapper

**Pull request merged**